### PR TITLE
Tic Tac Toe Javascript project

### PR DESCRIPTION
**Pull request by hillarioh**
Hi, this is our submission for code review of the tic tac toe javascript project. It entails 
- Form inputs
- Objects (Player and Board)
- Dom elements

We are looking forward to your review, Thanks

### CODE REVIEW

File: Js/index.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+const winningCases = [
+  [0, 1, 2],
+  [0, 4, 8],
+  [0, 3, 6],
+  [1, 4, 7],
+  [2, 5, 8],
+  [2, 4, 6],
+  [3, 4, 5],
+  [6, 7, 8],
+];
```
Comment (bellom) [inline Js/index.js]: I think all these can be called global variables right?
In the `Odin` instructions, you have been told to minimize the global variables in this project.
Note: Once you make use of `Factory Functions` and `The Module Pattern` you don't need to have any global variables in your code.

image

File: Js/index.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+}
+function checkWin(b) {
+  const w = winningCases;
+  for (let i = 0; i < w.length; i++) {
+    if (b.includes(w[i][0]) && b.includes(w[i][1]) && b.includes(w[i][2])) {
+      return true;
+    }
+  }
+}
+
```
Comment (bellom) [inline Js/index.js]: According to the instructions from `Odin` you are supposed to have the following classes `game`, `player`, and `gameboard`, don't forget to make use of factory functions.
image
https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript

File: Js/index.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+myForm.addEventListener('submit', (e) => {
+  e.preventDefault();
+  players.p1 = input1.value || 'Player 1';
+  players.p2 = input2.value || 'Player 2';
+  contentField.style.display = 'flex';
+  myForm.style.display = 'none';
+  ply1.innerHTML = players.p1;
+  ply2.innerHTML = players.p2;
+});
+
```
Comment (bellom) [inline Js/index.js]: Let's separate the DOM from application logic, you can create another file for DOM.

image

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

**Important Notice:** Don't let us forget the goal of this project, we need to make use of `Factory Functions` and `The Module Pattern` 
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59623090/

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by suretrust — APPROVED** [review]
Hi team, 👋 

### Great job implementing the suggested changes. 🙌 Let's get this merged! 👍 

- [ ] `OPTIONAL:` Your game should show when it's a draw.

_Good luck with your future projects and happy coding!_ 🎉

**Pull request merged**